spec.bs

```diff
@@ -759,6 +787,194 @@ in the [[#nested-traversables-intro]].
+  algorithm to read:
+
+  * User agents may focus one of those elements in the process, by running the
+    [=focusing steps=] for that element, and may change the scrolling position of the
+    document, or perform some other action that brings the element to the user's attention.
+    If these steps were invoked by user gesture, [=focusing steps=] can be called with
+    [=focus-unfenced|unfenced=] set to true. For elements that are
+    [=form-associated custom elements=], user agents should use their [=face validation anchor=]
+    instead, for the purposes of these actions.
+</div>
+
+  <h4 id=has-focus-steps-changes>Changes to the [=has focus steps=] algorithm</h4>
+
+  Add a step after step 2 of the while loop that reads:
+
+  3. If the [=focused area=] of
+    |candidate| is a [=fenced navigable container=] with a non-null
+    [=fenced navigable container/fenced navigable=], then set |candidate| to the
+    [=navigable/active document=] of that [=fenced navigable container=]'s
+    [=fenced navigable container/fenced navigable=].
+
+  <h4 id=focus-chain-changes>Changes to the [=focus chain=] algorithm</h4>
+
+  Modify step 3 of the while loop in the [=focus chain=] algorithm to read:
+  3. If |currentObject| is a [=focusable area=], then set
+    |currentObject| to |currentObject|'s [=DOM anchor=]'s 
+    [=Node/node document=].
+
+    Otherwise, if |currentObject| is a {{Document}} whose
+    [=node navigable=]'s [=navigable/parent=] is non-null, then set
+    |currentObject| to |currentObject|'s [=node navigable=]'s [=navigable/parent=].
+
+    Otherwise, if |currentObject| is a {{Document}} whose
+    [=node navigable=]'s [=traversable navigable/unfenced parent=] is non-null, then set
+    |currentObject| to |currentObject|'s [=node navigable=]'s
+    [=traversable navigable/unfenced parent=].
+
+    Otherwise, <span>break</span>.
+
+  <h4 id=focusable-area-changes>Changes to the [=get the focusable area=] algorithm</h4>
+
+  Add a new case to the switch statement:
+
+  <dl class="switch">
+    <dt>If <var ignore>focus target</var> is a [=fenced navigable container=] with a non-null
+    [=fenced navigable container/fenced navigable=]</dt>
+
+    <dd><p>Return the [=fenced navigable container=]'s
+    [=fenced navigable container/fenced navigable=]'s [=navigable/active document=].</p></dd>
+  </dl>
+
+  <h4 id=currently-focused-area-changes>Changes to the [=currently focused area of a top-level traversable=] algorithm</h4>
+
+  Modify step 3 of the [=currently focused area of a top-level traversable=] algorithm to read:
+
+  3. While |candidate|'s [=focused area=] is either a [=navigable container=] with a non-null
+    [=navigable container/content navigable=] or a [=fenced navigable container=] with a non-null
+    [=fenced navigable container/fenced navigable=]: set |candidate| to the
+    [=navigable/active document=] of either that [=navigable container=]'s
```

Comment:
I think the last couple lines here could be simplified with https://wicg.github.io/fenced-frame/#navigable-unfenced-parent that I added recently. Basically not only does a traversable navigable have the concept of an unfenced parent (member), but now there is also a `[=navigable/unfenced parent=]` _algorithm_ that conditionally takes advantage of the member.

Reply:
This algorithm is traversing from the top-level down, not from the bottom up. What I really need is something like an "unfenced child" algorithm. Is that something that would be worth adding to the spec?

Reply:
Yes I think that would be useful. Feel free to post a separate PR for that if you'd like, but no need to block this one.